### PR TITLE
DCMAW-7076:Android | Initiate login

### PR DESCRIPTION
**Pull request by MohamedHassan101**
### DCMAW-7076:Android | Initiate login
- Update url build with the correct query parameters
- Update url staging with the correct query parameters

Video evidence for **Staging** 

https://github.com/govuk-one-login/mobile-android-one-login-app/assets/113670948/a68d4003-db43-4181-b5a0-9b33fefb3f52

### CODE REVIEW

File: app/src/staging/res/values/login_flow.xml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <string name="openIdConnectBaseUrl" translatable="false">https://oidc.staging.account.gov.uk%1$s</string>
-    <string name="openIdConnectClientId" translatable="false">CLIENT_ID</string>
-    <string name="webBaseUrl" translatable="false">https://mobile-staging.account.gov.uk%1$s</string>
-</resources>
\ No newline at end of file
+    <string name="openIdConnectBaseUrl" translatable="false">https://oidc.integration.account.gov.uk%1$s</string>
+    <string name="openIdConnectClientId" translatable="false">sdJChz1oGajIz0O0tdPdh0CA2zW</string>
```
Comment (Chuddington) [inline app/src/staging/res/values/login_flow.xml:4]: I don't know if it matters so I'll ask anyway - are we happy with having this hardcoded within the strings / public code base?

**Review by Chuddington — APPROVED** [review]
Looks good. Got a comment on the Client ID

File: app/src/staging/res/values/login_flow.xml (hunk 1 above)
Comment (cgedrim-co) [reply app/src/staging/res/values/login_flow.xml:4]: @elishkaflint one for you maybe?

Comment (elishkaflint) [reply app/src/staging/res/values/login_flow.xml:4]: Thanks for flagging @Chuddington! I'm checking and will come back on this as soon as possible. I think the client id is a public identifier and not considered a secret per the spec (there are other controls eg. the client assertion - signed with a secret private key - during the token exchange) but let me double-check that.

File: app/src/main/java/uk/gov/onelogin/login/UriBuilder.kt [hunk 2]
```diff
@@ -12,7 +12,7 @@ data class UriBuilder(
 ) {
   val url: Uri = Uri.parse(baseUri)
             .buildUpon().appendQueryParameter("response_type", "code")
-            .appendQueryParameter("scope", "openid email phone offline_access")
+            .appendQueryParameter("scope", "openid offline_access")
```
Comment (elishkaflint) [inline app/src/main/java/uk/gov/onelogin/login/UriBuilder.kt:15]: My bad - but the ticket was incorrect - the scope should just be `openid`

We corrected this when we did the integration between iOS and the stub but I forgot to go back and update the Android tickets. 

That's done now, please can we update this accordingly? Thank you and sorry for catching this so late!

Comment (elishkaflint) [reply app/src/main/java/uk/gov/onelogin/login/UriBuilder.kt:15]: Screenshot 2023-11-08 at 16 14 21

File: app/src/staging/res/values/login_flow.xml (hunk 1 above)
Comment (elishkaflint) [reply app/src/staging/res/values/login_flow.xml:4]: Confirmed that this is not considered a secret here https://gds.slack.com/archives/C04L0BZQNR3/p1699459522702629

Please feel free to resolve this convo once you've seen it and are happy @MohamedHassan101

File: app/src/main/java/uk/gov/onelogin/login/UriBuilder.kt (hunk 2 above)
Comment (MohamedHassan101) [reply app/src/main/java/uk/gov/onelogin/login/UriBuilder.kt:15]: no problem , will update that now

File: app/src/staging/res/values/login_flow.xml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <string name="openIdConnectBaseUrl" translatable="false">https://oidc.staging.account.gov.uk%1$s</string>
-    <string name="openIdConnectClientId" translatable="false">CLIENT_ID</string>
-    <string name="webBaseUrl" translatable="false">https://mobile-staging.account.gov.uk%1$s</string>
-</resources>
\ No newline at end of file
+    <string name="openIdConnectBaseUrl" translatable="false">https://oidc.integration.account.gov.uk%1$s</string>
```
[the remark this answers is not in the document]
Comment (MohamedHassan101) [reply app/src/staging/res/values/login_flow.xml:3]: This is used to appending one string to the end of another string.In this case we are adding "/authorize" at the end of https://oidc.integration.account.gov.uk%1$s

File: app/src/build/res/values/login_flow.xml [hunk 4]
```diff
@@ -1,6 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <string name="openIdConnectBaseUrl" translatable="false">https://oidc.build.account.gov.uk%1$s</string>
-    <string name="openIdConnectClientId" translatable="false">CLIENT_ID</string>
+    <string name="openIdConnectBaseUrl" translatable="false">https://auth-stub.mobile.dev.account.gov.uk%1$s</string>
+    <string name="openIdConnectClientId" translatable="false">TEST_CLIENT_ID</string>
     <string name="webBaseUrl" translatable="false">https://mobile-build.account.gov.uk%1$s</string>
+
```
Comment (elishkaflint) [inline app/src/build/res/values/login_flow.xml:6]: can we remove these blank lines?

File: app/src/staging/res/values/login_flow.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (cgedrim-co) [reply app/src/staging/res/values/login_flow.xml:3]: It's an `printf`-ish placeholder, so no crazy encoding or anything 😉 

You'd use it like this:

```
<string name="example">Hello %1$s, good %2$s</string>
```

```
resources.getString(
  R.string.example,
  "Elishka",
  "evening"
)
```

Which would render "Hello Elishka, good evening"

**Review by elishkaflint — APPROVED** [review]

**Pull request merged**